components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1743,6 +1790,7 @@ where
         self.fsm.missing_ticks = 0;
         self.fsm.peer.should_wake_up = false;
         self.register_raft_base_tick();
+        self.register_check_leader_lease_tick();
```

Comment:
It should be registered only when it becomes leader.

Reply:
And should it be registered after being awakened?

Reply:
I would suggest to disable the tick if hibernate region is enabled as the waken request will extends the lease any way.

Reply:
The duration before the region goes to hibernate would be very long when some stores were isolated.

Reply:
Then here should check if it's leader before registering the tick.

Reply:
The cost is acceptable according to our tests.